client/state/selectors/should-show-launchpad-first.ts

```diff
@@ -3,15 +3,18 @@ import { SiteDetails, Onboard } from '@automattic/data-stores';
 const SiteIntent = Onboard.SiteIntent;
 
 /**
- * Determines if the launchpad should be shown first based on site createion flow
+ * Determines if the launchpad should be shown first based on site creation flow.
  * @param site Site object
  * @returns Whether launchpad should be shown first
  */
 export const shouldShowLaunchpadFirst = ( site: SiteDetails ): boolean => {
 	const wasSiteCreatedOnboardingFlow = site.options?.site_creation_flow === 'onboarding';
 	const isBigSkyIntent = site?.options?.site_intent === SiteIntent.AIAssembler;
+	// If we don't have a site intent, fall through to the next option.
+	const siteHasNoIntent =
+		site && site.options && ( site.options.site_intent === '' || ! site.options.site_intent );
```

Comment:
Rather than testing for different kinds of emptiness, maybe we could flip the sense of the `! wasSiteCreatedOnboardingFlow`. That is, remove the early exit and replace the functions logic with:

```ts
return wasSiteCreatedOnboardingFlow && ! isBigSkyIntent
```

Reply:
Thanks for the review, @p-jackson!

I don't think that would address this issue, TBH, as your proposed condition would still return `true` for the case where the site _was_ created via the `onboarding` flow, but doesn't have any declared intent.

Reply:
Woops you're right. I still think flipping the sense makes things easier though:

`isOnboardingFlow && hasIntent && ! bigSky`

> I am also happy for you and your team to consider how to handle this better in general 

I'm eod so I'm happy for you to merge this, it doesn't really look like a bug we should delay fixing.

Reply:
Agreed! I'd be happy for the logic to get cleaned up, but I wanted to fix the immediate problem as an initial priority. So I've merged this as-is, and we can improve from here.